include/s3/checksum.h

```diff
@@ -0,0 +1,52 @@
+/*-------------------------------------------------------------------------
+ *
+ * checksum.h
+ * 		Declarations for calculating checksums of S3-specific data.
+ *
+ * Copyright (c) 2024, Oriole DB Inc.
+ *
+ * IDENTIFICATION
+ *	  contrib/orioledb/include/utils/checksum.h
+ *
+ *-------------------------------------------------------------------------
+ */
+#ifndef __S3_CHECKSUM_H__
+#define __S3_CHECKSUM_H__
+
+#include "utils/hsearch.h"
+
+#include "openssl/sha.h"
+
+#define O_SHA256_DIGEST_STRING_LENGTH	(SHA256_DIGEST_LENGTH * 2 + 1)
+
+typedef struct S3FileChecksum
+{
+	char		filename[MAXPGPATH];
+	char		checksum[O_SHA256_DIGEST_STRING_LENGTH];
+
+	bool		changed;		/* true if crc changed since last checkpoint */
+	uint32		checkpointNumber;
+} S3FileChecksum;
+
+typedef struct S3ChecksumState
+{
+	HTAB	   *hashTable;
+	uint32		checkpointNumber;
+
+	S3FileChecksum *pgFiles;		/* Buffer of S3FileChecksum entries */
```

Comment:
Maybe renane pgFiles -> fileChecksums/pgFileChecksums 
Same for pgFilesMaxLen, pgFilesLen,  WORKERS_PGFILES_MAX_LEN, pgFilesCnt, pgFilesFlushedCV, PGFILES_MAX_LEN, worker_pgfiles etc.

Reply:
Done